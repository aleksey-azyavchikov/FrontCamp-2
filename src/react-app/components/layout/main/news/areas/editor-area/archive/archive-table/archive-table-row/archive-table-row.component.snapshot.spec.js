import ArchiveTableRow from "./archive-table-row.component";
import renderer from "react-test-renderer";
import React from "react";

describe("Archive table row", () => {
    let article;
    let index;
    beforeAll(() => {
        article = {
            title: "Title",
            description: "Description",
            publishedAt: "3-2-2018",
            url: "http://test.com",
        };
        index = 5;
    })
    it("Display correctly", () => {
        const filter = "test";
        const rendered = renderer.create(
            <ArchiveTableRow 
                article={article}
                index={index}
                filter={filter}
            />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    })
    
    it("Display correctly with highliting", () => {
        const filter = "ti";
        const rendered = renderer.create(
            <ArchiveTableRow 
                article={article}
                index={index}
                filter={filter}
            />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    })
}) 
