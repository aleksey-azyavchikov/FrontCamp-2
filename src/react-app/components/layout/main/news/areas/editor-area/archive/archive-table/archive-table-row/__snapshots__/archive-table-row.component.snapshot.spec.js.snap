// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Archive table row Display correctly 1`] = `
<tr
  className="article"
  onClick={[Function]}
>
  <th
    scope="row"
  >
    5
  </th>
  <td>
    <span
      className={undefined}
    >
      <span
        className=""
        style={undefined}
      >
        Title
      </span>
    </span>
  </td>
  <td>
    Description
  </td>
  <td>
    3-2-2018
  </td>
  <td>
    <a
      href="http://test.com"
    >
      Link
    </a>
  </td>
</tr>
`;

exports[`Archive table row Display correctly with highliting 1`] = `
<tr
  className="article"
  onClick={[Function]}
>
  <th
    scope="row"
  >
    5
  </th>
  <td>
    <span
      className={undefined}
    >
      <mark
        className=" "
        style={
          Object {
            "fontWeight": "normal",
          }
        }
      >
        Ti
      </mark>
      <span
        className=""
        style={undefined}
      >
        tle
      </span>
    </span>
  </td>
  <td>
    Description
  </td>
  <td>
    3-2-2018
  </td>
  <td>
    <a
      href="http://test.com"
    >
      Link
    </a>
  </td>
</tr>
`;
